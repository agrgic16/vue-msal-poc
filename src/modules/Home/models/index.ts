export * from './user-info-state.model';
