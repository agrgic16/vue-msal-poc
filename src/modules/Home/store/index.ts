import Vue from 'vue';
import Vuex, { StoreOptions, Store, ActionContext } from 'vuex';
import { AxiosResponse } from 'axios';
import * as Msal from 'msal';

import microsoftAPI from '@/core/api/microsoft.api';
import { UserInfoState } from '../models';

Vue.use(Vuex);

export enum Mutations {
  LoginPending = 'Auth/LOGIN_PENDING',
  LoginSuccess = 'Auth/LOGIN_SUCCESS',
  LoginError = 'Auth/LOGIN_ERROR',

  FetchUserInfoPending = 'Auth/FETCH_USER_INFO_PENDING',
  FetchUserInfoSuccess = 'Auth/FETCH_USER_INFO_SUCCESS',
  FetchUserInfoError = 'Auth/FETCH_USER_INFO_ERROR',
}

export const storeOptions: StoreOptions<UserInfoState> = {
  strict: process.env.NODE_ENV !== 'production',
  state: {
    userInfo: {
      data: null,
      isLoading: false,
      error: null,
    },
  },
  mutations: {
    [Mutations.FetchUserInfoPending](state: UserInfoState) {
      state.userInfo = {
        data: null,
        isLoading: true,
        error: null,
      };
    },
    [Mutations.FetchUserInfoSuccess](state: UserInfoState, { data }: AxiosResponse<any>) {
      state.userInfo = {
        ...state.userInfo,
        data,
        isLoading: false,
      };
    },
    [Mutations.FetchUserInfoError](state: UserInfoState, error: any) {
      state.userInfo = {
        data: null,
        isLoading: false,
        error,
      };
    },
  },
  actions: {
    async getUserInfo({ commit }: ActionContext<UserInfoState, UserInfoState>) {
      try {
        commit(Mutations.FetchUserInfoPending);
        commit(Mutations.FetchUserInfoSuccess, await microsoftAPI.get('/v1.0/me'));
      } catch (err) {
        commit(Mutations.FetchUserInfoError, err);
      }
    },
  },
  getters: {
    isLoading(state: UserInfoState): boolean {
      return state.userInfo.isLoading;
    },
    isError(state: UserInfoState): boolean {
      return !!state.userInfo.error;
    },
    userInfo(state: UserInfoState): any {
      return state.userInfo.data;
    },
  },
};

export default new Vuex.Store(storeOptions);
