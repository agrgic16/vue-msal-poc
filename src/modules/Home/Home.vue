<template>
  <v-layout column>
    <v-flex xs10 sm2 offset-sm5>
      <v-card
          >
        <v-img
          class="image"
          src="https://media.licdn.com/dms/image/C4D0BAQGe0QkrWz6VIQ/company-logo_200_200/0?e=2159024400&v=beta&t=E8D4ycI_q4_cr9qGS88OWyZ8diZcgV0PtbqsNpz2nOE"
          aspect-ratio="2.75"
        ></v-img>

        <v-card-title primary-title>
          <div>
            <h3 class="headline mb-0">Career Paths Project</h3>
          </div>
        </v-card-title>

        <v-card-actions>
          <v-btn text color="blue">Button</v-btn>
          <v-btn text color="blue">Button</v-btn>
        </v-card-actions>
      </v-card>
    </v-flex>
    <v-card>
      <pre>
        <code>{{userInfo}}</code>
      </pre>
    </v-card>
  </v-layout>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { mapGetters, mapActions } from 'vuex';
import { Route } from 'vue-router';

import store, { Mutations } from '@/modules/Home/store';

@Component<Home>({
  components: {
  },
  store,
  computed: {
    ...mapGetters(['isLoggedIn', 'isLoading', 'isError', 'userInfo']),
  },
  methods: {
    ...mapActions(['getUserInfo']),
  },
})
export default class Home extends Vue {
  isLoggedIn!: boolean;

  isLoading!: boolean;

  isError!: boolean;

  userInfo!: any;

  getUserInfo!: () => void;

  created() {
    this.getUserInfo();
  }
}
</script>

<style lang="scss">
  .image {
    height: 280px;
  }
</style>
