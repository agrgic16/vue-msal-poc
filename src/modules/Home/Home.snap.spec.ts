import Vue from 'vue';
import vueRouter from 'vue-router';
import Vuetify from 'vuetify';
import { shallowMount } from '@vue/test-utils';
import Home from './Home.vue';

Vue.use(Vuetify, {
  iconfont: 'md',
});

describe('App.vue component matches snapshot', () => {
  it('for default view', () => {
    Vue.use(vueRouter);
    const wrapper = shallowMount(Home);
    expect(wrapper).toMatchSnapshot();
  });
});
