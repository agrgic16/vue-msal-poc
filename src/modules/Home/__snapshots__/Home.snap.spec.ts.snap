// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue component matches snapshot for default view 1`] = `
<v-layout-stub tag="div">
  <v-flex-stub tag="div" xs10="" sm2="" offset-sm5="">
    <v-card-stub tag="div">
      <v-img-stub aspectratio="2.75" src="https://media.licdn.com/dms/image/C4D0BAQGe0QkrWz6VIQ/company-logo_200_200/0?e=2159024400&amp;v=beta&amp;t=E8D4ycI_q4_cr9qGS88OWyZ8diZcgV0PtbqsNpz2nOE" position="center center" transition="fade-transition" class="image"></v-img-stub>
      <v-card-title-stub primarytitle="true">
        <div>
          <h3 class="headline mb-0">Career Paths Project</h3>
        </div>
      </v-card-title-stub>
      <v-card-actions-stub>
        <v-btn-stub color="blue" activeclass="v-btn--active" tag="button" flat="true" type="button">Button</v-btn-stub>
        <v-btn-stub color="blue" activeclass="v-btn--active" tag="button" flat="true" type="button">Button</v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-flex-stub>
</v-layout-stub>
`;
